os/emscripten/cmake/FindLibLZMA.cmake

```diff
@@ -0,0 +1,19 @@
+# LibLZMA is a recent addition to the emscripten SDK, so it is possible
+# someone hasn't updated his SDK yet. Test out if the SDK supports LibLZMA.
+include(CheckCXXSourceCompiles)
+set(CMAKE_REQUIRED_FLAGS "-sUSE_LIBLZMA=1")
+
+check_cxx_source_compiles("
+    int main() { return 0; }"
+    LIBLZMA_FOUND
+)
```

Comment:
This doesn't appear to actually test lzma? Shouldn't it at least be trying to include the header? Alternatively, is it necessary at all? None of the other emscripten Find*.cmake files have compile tests

Reply:
> None of the other emscripten Find*.cmake files have compile tests
FYI, we do this in some other `Find` scripts too, like the `Xaudio2` :) So I did not invent something new :D (I might have copied it from there :P)

Nevertheless, it is a bit uncommon, and for this you need to know a bit about how emscripten works.

So, emscripten has a very unusual way of  installing dependencies, or "ports" in their lingo. All those `-s USE_NNN` tell the wrapping scripts to ensure that dependency exists on the system. So if we do `-sUSE_LIBLZMA=1`, and compile anything with that, no matter what, emscripten is going to download and install the port of LibLZMA, fully transparent and automated.

Honestly, this is one of the more weird things emscripten does. It is also the reason you don't see us installing any dependencies, and why we need to have a patch to install a port. Rather I would have seen a command that installs the dependency of course, like all other systems, but alas, here we are :) Welcome to the world of emscripten :D

As far as this piece of code, I could add an `#include` to make it look a bit less weird. It is still weird :) I didn't want to add comments about how emscripten works, as .. that is like documenting how GCC works. But I think with adding an `#include` everyone is fine with it :)